tests/utils.py

```diff
@@ -0,0 +1,22 @@
+import socket
+from random import randint
+from typing import Set
+
+max_tries = 10
+
+
+def get_random_unused_port(allocated_ports: Set = set()):
```

Comment:
This default argument is mutable by design and is necessary for the function to exist correctly. Honestly it seems like I'm trolling but I can't think of a better way to maintain this function's state between function calls (possibly because abusing this counterintuitive behaviour fills me with glee).

Reply:
an evil more terrible way to do this would be to keep an unencrypted text file that describes the state.